### PR TITLE
haiku build fix proposal.

### PR DESCRIPTION
**Pull request by devnexen**
Disabling non existing features and flags (close on exec). Fixing ControlLen type.

### CODE REVIEW

**Review by tormol — CHANGES_REQUESTED** [review]
Looks good.
Is there a rustup target for haiku? (to check that it builds).

File: src/helpers.rs [hunk 1]
```diff
@@ -42,6 +44,7 @@ pub const MSG_NOSIGNAL: c_int = 0; // SO_NOSIGPIPE is set instead
 
 
 /// Enables / disables CLOEXEC, for when SOCK_CLOEXEC can't be used.
+#[cfg(not(target_os = "haiku"))]
```
[the remark this answers is not in the document]
Comment (devnexen) [reply src/helpers.rs:47]: works for me.

**Comment by devnexen** [comment]
> Looks good. Is there a rustup target for haiku? (to check that it builds).

no unfortunately

**Review by tormol — APPROVED** [review]

**Comment by tormol** [comment]
I've fixed the MSRV and macOS build failures on master. You just need to rebase.

**Comment by tormol** [comment]
This is finally released! in 0.3.0.

**Comment by devnexen** [comment]
Hey ! I forgot about this one :)

**Pull request merged**